src/caret_analyze/runtime/path.py

```diff
@@ -214,17 +215,33 @@ def is_match_column(column: str, target_name: str) -> bool:
             first_element = targets[0].to_path_beginning_records()
         else:
             if len(targets[0].to_records().data) == 0:
+                # If the first element has no data, skip it and re-evaluate targets
                 targets = targets[1:]
-
+                if not targets:  # Check if targets list became empty after removal
+                    msg = (
+                        'Targets list became empty after filtering out first element. '
+                        'Returning empty records.'
+                    )
+                    raise InvalidRecordsError(msg)
             first_element = targets[0].to_records()
 
-        left_records = first_element
-
-        rename_rule = column_merger.append_columns_and_return_rename_rule(
-            left_records)
+        left_records: RecordsInterface = first_element
 
+        rename_rule = column_merger.append_columns_and_return_rename_rule(left_records)
         left_records.rename_columns(rename_rule)
-        first_column = first_element.columns[0]
+        # Handle case where first_element might have no columns
+        first_column = left_records.columns[0] if left_records.columns else None
+
+        last_communication_record = None
+        for item in reversed(targets):
+            if isinstance(item, Communication):
+                last_communication_record = item
+                break
+
+        if not last_communication_record:
+            raise InvalidRecordsError('Communication record does not exist')
```

Comment:
last_communication_record is of type Communication, not RecordsInterface, so the variable name might be misleading.